### PR TITLE
Add str2e function

### PR DESCRIPTION
**Pull request by wangch079**
This PR adds a `str2e` function in module `connectors.utils`, to convert a string to an Enum value

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

**Comment by tarekziade** [comment]
related: @vidok proposed to move Enums to strings, 

e.g.

```
class Status(Enum):
    CREATED = 1
    NEEDS_CONFIGURATION = 2
    CONFIGURED = 3
    CONNECTED = 4
    ERROR = 5
```

becomes

```
class Status(Enum):
    CREATED = 'created'
    NEEDS_CONFIGURATION = 'needs_configuration'
    CONFIGURED = 'configured'
    CONNECTED = 'connected'
    ERROR = 'error'
```

ISTM that this would make our lives easier and avoid extra conversions

File: connectors/utils.py [hunk 1]
```diff
@@ -401,3 +401,14 @@ async def func_to_execute(*args, **kwargs):
 
 def e2str(entry):
     return entry.name.lower()
+
+
+def str2e(name, enum_class):
+    if name is None:
```
Comment (tarekziade) [inline connectors/utils.py:407]: it seems that this function will return `None` when the name is not know. Shouldn't we raise an error instead? I can think of problems where None will be used silently and can't think of scenarios where we ask for a name that is not supposed to exist in the Enum

**Comment by wangch079** [comment]
>related: @vidok proposed to move Enums to strings,

I think a `str2e` function is still needed?

We can use `Status.CREATED.value` instead of `e2str` but I don't think this is much difference.

**Comment by tarekziade** [comment]
> 



> > related: @vidok proposed to move Enums to strings,
> 
> I think a `str2e` function is still needed?

no because you can then do:

```
Status('created')
```

> 
> We can use `Status.CREATED.value` instead of `e2str` but I don't think this is much difference.

File: connectors/utils.py (hunk 1 above)
Comment (tarekziade) [reply connectors/utils.py:407]: Gotcha. I guess if we use strings in the enums we can have an explicit set:

```
enum_value = None if value is None else JobStatus(value)
```

Comment (tarekziade) [reply connectors/utils.py:407]: or an unset value:

```
>>> class MyEnum(Enum):
...     one = 'one'
...     two = 'two'
...     unset = None
...
>>> MyEnum(None)
<MyEnum.unset: None>
>>> MyEnum('one')
<MyEnum.one: 'one'>
```

**Review by tarekziade — APPROVED** [review]
please consider all my feedback as possible improvements, but the new function is good enough if that helps you to unblock other PRS

**Pull request merged**